### PR TITLE
Add LoginScreen, Screen Coordinators.

### PR DESCRIPTION
**Pull request by sxcore**
Please review my changes. Whole project structure has been updated.

### CODE REVIEW

File: ELSpace.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -9,10 +9,16 @@
...
+		500BC9771EE7EF920060D31B /* ScreenBase.swift in Sources */ = {isa = PBXBuildFile; fileRef = 500BC9761EE7EF920060D31B /* ScreenBase.swift */; };
+		500BC97C1EE7F7380060D31B /* ScreenCoordinator.swift in Sources */ = {isa = PBXBuildFile; fileRef = 500BC97B1EE7F7380060D31B /* ScreenCoordinator.swift */; };
 		5048F2441EDEE7A2001A487E /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5048F2431EDEE7A2001A487E /* AppDelegate.swift */; };
 		5048F2461EDEE7A2001A487E /* MainViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5048F2451EDEE7A2001A487E /* MainViewController.swift */; };
 		5048F24B1EDEE7A2001A487E /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 5048F24A1EDEE7A2001A487E /* Assets.xcassets */; };
 		5048F2591EDEE7A2001A487E /* ELSpaceTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5048F2581EDEE7A2001A487E /* ELSpaceTests.swift */; };
+		5063AE191EE7003A00D4E45A /* R.generated.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5063AE181EE7003A00D4E45A /* R.generated.swift */; };
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: R.generated should not be commited to the repo. Only generated and used locally.

File: ELSpace.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -36,6 +44,10 @@
 		5048F2541EDEE7A2001A487E /* ELSpaceTests.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = ELSpaceTests.xctest; sourceTree = BUILT_PRODUCTS_DIR; };
 		5048F2581EDEE7A2001A487E /* ELSpaceTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ELSpaceTests.swift; sourceTree = "<group>"; };
 		5048F25A1EDEE7A2001A487E /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
+		5063AE181EE7003A00D4E45A /* R.generated.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = R.generated.swift; sourceTree = "<group>"; };
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: As above

File: ELSpace.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -113,45 +133,59 @@
...
+		};
 		509E0D751EDEE9B700414C1D /* Supporting FIles */ = {
 			isa = PBXGroup;
 			children = (
+				50902FC91EE5AD5200D43254 /* Translations */,
 				5048F24A1EDEE7A2001A487E /* Assets.xcassets */,
-				5048F24F1EDEE7A2001A487E /* Info.plist */,
 				866AADE51EE1A8BA00FAEB6D /* ElSpaceLaunchScreen.storyboard */,
+				5048F24F1EDEE7A2001A487E /* Info.plist */,
+				5063AE181EE7003A00D4E45A /* R.generated.swift */,
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: As above

File: ELSpace.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -113,45 +133,59 @@
...
 			);
-			name = "Supporting FIles";
+			path = "Supporting FIles";
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: Typo "..FIles"

File: ELSpace.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -113,45 +133,59 @@
...
 			sourceTree = "<group>";
 		};
 		509E0D771EDEEA0600414C1D /* Main View Controller */ = {
 			isa = PBXGroup;
 			children = (
+				50902FC51EE5AAC700D43254 /* MainView.swift */,
 				5048F2451EDEE7A2001A487E /* MainViewController.swift */,
 			);
-			name = "Main View Controller";
+			path = "Main View Controller";
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: IMO remove whitespaces - > MainViewController

File: ELSpace.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -368,7 +418,12 @@
 			isa = PBXSourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				500BC9771EE7EF920060D31B /* ScreenBase.swift in Sources */,
 				5048F2461EDEE7A2001A487E /* MainViewController.swift in Sources */,
+				50902FC61EE5AAC700D43254 /* MainView.swift in Sources */,
+				5063AE191EE7003A00D4E45A /* R.generated.swift in Sources */,
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: As above

File: ELSpace.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -493,7 +560,7 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer: Dariusz Rybicki (KB57E8RV46)";
 				CURRENT_PROJECT_VERSION = 6;
 				DEVELOPMENT_TEAM = H5QFB36W2F;
-				INFOPLIST_FILE = ELSpace/Info.plist;
+				INFOPLIST_FILE = "ELSpace/Supporting FIles/Info.plist";
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: Same typo

File: ELSpace.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -512,7 +579,7 @@
...
```
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: typo

File: ELSpace/Coordinators/ScreenCoordinator.swift [hunk 9]
```diff
@@ -0,0 +1,22 @@
+//
+//  Created by Michał Czerniakowski on 07.06.2017.
+//Copyright © 2017 El Passion. All rights reserved.
+//
+
+import UIKit
+
+class ScreenCoordinator {
```
Comment (bartguminiak) [inline ELSpace/Coordinators/ScreenCoordinator.swift]: How about AppCoordinator ?

File: ELSpace.xcodeproj/project.pbxproj (hunk 5 above)
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: And how about renaming it to LoginViewController as far it is login screen?

File: ELSpace.xcodeproj/project.pbxproj (hunk 7 above)
Comment (bartguminiak) [inline ELSpace.xcodeproj/project.pbxproj]: typo

File: ELSpace/Screens/Elements/ScreenBase.swift [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        configureAutolayout()
+    }
+    
+    private func configureBackgroundGradient() {
+        gradientView.startPastelPoint = .topLeft
+        gradientView.endPastelPoint = .bottomRight
+        gradientView.animationDuration = 3.5
+        
+        gradientView.setColors([
+            UIColor("#17EAD9")!,
```
Comment (bartguminiak) [inline ELSpace/Screens/Elements/ScreenBase.swift]: "!" we can perform more safely

File: ELSpace/Screens/Main View Controller/MainViewController.swift [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import HexColors
+
+class MainViewController: UIViewController {
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+    
+    override func loadView() {
+        view = MainView()
```
Comment (bartguminiak) [inline ELSpace/Screens/Main View Controller/MainViewController.swift]: How about LoginView?

File: ELSpace/Screens/Elements/ButtonBase.swift [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        configureButtonLayout()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func configureButtonLayout() {
+        setTitleColor(UIColor.white, for: .normal)
+        titleLabel?.font = UIFont(name: "HelveticaNeue", size: 16.0)
```
Comment (bartguminiak) [inline ELSpace/Screens/Elements/ButtonBase.swift]: Use Rswift to keep it typed.

File: ELSpace/Screens/Elements/ScreenBase.swift (hunk 10 above)
Comment (bartguminiak) [reply ELSpace/Screens/Elements/ScreenBase.swift]: Rswift may help. https://github.com/mac-cain13/R.swift/blob/master/Documentation/Colors.md

File: ELSpace/Screens/Main View Controller/MainViewController.swift [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    
+    private var mainView: MainView {
+        guard let mainView = view as? MainView else { fatalError("Expected LoginView but got \(type(of: view))") }
```
Comment (bartguminiak) [inline ELSpace/Screens/Main View Controller/MainViewController.swift]: Copy-pasted ?

File: ELSpace/Screens/Supporting Files/Translations/Base.lproj/Localizable.strings [hunk 14]
```diff
@@ -0,0 +1,7 @@
+/* 
+  Created by Michał Czerniakowski on 05.06.2017.
+  Copyright © 2017 El Passion. All rights reserved.
+*/
+
+"google_login_button" = "Login With Google Account";
+"provide_access_token" = "Provide Your Access Token";
```
Comment (bartguminiak) [inline ELSpace/Screens/Supporting Files/Translations/Base.lproj/Localizable.strings]: At the planning session we agreed to log in only with google account. Remove direct token access flow.

**Review by bartguminiak — CHANGES_REQUESTED** [review]

File: ELSpace/Screens/Main View Controller/MainViewController.swift (hunk 13 above)
Comment (sxcore) [reply ELSpace/Screens/Main View Controller/MainViewController.swift]: I meant Login View 🗡

File: ELSpace/Screens/Main View Controller/MainViewController.swift (hunk 11 above)
Comment (sxcore) [reply ELSpace/Screens/Main View Controller/MainViewController.swift]: Changed 🍡

File: ELSpace/Screens/Supporting Files/Translations/Base.lproj/Localizable.strings (hunk 14 above)
Comment (sxcore) [reply ELSpace/Screens/Supporting Files/Translations/Base.lproj/Localizable.strings]: Deleted

File: ELSpace.xcodeproj/project.pbxproj (hunk 5 above)
Comment (sxcore) [reply ELSpace.xcodeproj/project.pbxproj]: Renamed already 📟

File: ELSpace/Coordinators/ScreenCoordinator.swift (hunk 9 above)
Comment (sxcore) [reply ELSpace/Coordinators/ScreenCoordinator.swift]: ok, done 🔨

File: ELSpace/Screens/Elements/ButtonBase.swift (hunk 12 above)
Comment (sxcore) [reply ELSpace/Screens/Elements/ButtonBase.swift]: Done 🌵

File: ELSpace.xcodeproj/project.pbxproj (hunk 8 above)
Comment (sxcore) [reply ELSpace.xcodeproj/project.pbxproj]: done

File: ELSpace/Screens/Login/LoginViewController.swift [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        configureSubviews()
+    }
+
+    private func configureSubviews() {
+        view.backgroundColor = UIColor("1BCEDF")
```
Comment (bartguminiak) [inline ELSpace/Screens/Login/LoginViewController.swift]: Let LoginView be more consistent and move setting background color over there.

File: ELSpace/Coordinators/AppCoordinator.swift [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+    
+    fileprivate let window: UIWindow
+    
+    fileprivate let loginViewController = LoginViewController()
+    
+    init(window: UIWindow) {
+        self.window = window
+    }
+    
+    internal func present() {
```
Comment (bartguminiak) [inline ELSpace/Coordinators/AppCoordinator.swift]: Any good reason to use internal access here?

File: ELSpace/Screens/Login/LoginView.swift [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+//
+
+import UIKit
+import Anchorage
+import HexColors
+
+class LoginView: UIView {
+    
+    private let backgroundView = BackgroundGradientWithLogo()
+    internal let loginButton = Button(frame: .zero)
```
Comment (bartguminiak) [inline ELSpace/Screens/Login/LoginView.swift]: internal why ?

File: Podfile [hunk 18]
```diff
@@ -3,7 +3,7 @@ inhibit_all_warnings!
 
 def helpers
     pod 'Anchorage', '~> 4.0'
-    pod 'UIColor_Hex_Swift', '~> 3.0.2'
+    pod 'HexColors', '~> 5.0.1'
     pod "Pastel", '~> 0.3.0'
```
Comment (bartguminiak) [inline Podfile]: Use '' instead of ""

**Review by bartguminiak — CHANGES_REQUESTED** [review]

File: Podfile [hunk 19]
```diff
@@ -1,11 +1,24 @@
 platform :ios, '10.0'
+inhibit_all_warnings!
 
-target 'ELSpace' do
-  use_frameworks!
+def helpers
+    pod 'Anchorage', '~> 4.0'
+    pod 'HexColors', '~> 5.0.1'
+    pod "Pastel", '~> 0.3.0'
```
Comment (bartguminiak) [inline Podfile]: Use '' instead of ""

Comment (sxcore) [reply Podfile]: Done, -> mistake of copy paste 👎

File: ELSpace/Screens/Login/LoginViewController.swift (hunk 15 above)
Comment (sxcore) [reply ELSpace/Screens/Login/LoginViewController.swift]: Ok, my mistake. Fixed ASAP 🥇

File: ELSpace/Coordinators/AppCoordinator.swift (hunk 16 above)
Comment (sxcore) [reply ELSpace/Coordinators/AppCoordinator.swift]: Totally forgot about another type of encapsulation!

**Pull request merged**